pyfr/plugins/writer.py

```diff
@@ -1,14 +1,39 @@
+import os
+
 from pyfr.inifile import Inifile
 from pyfr.mpiutil import get_comm_rank_root
-from pyfr.plugins.base import BasePlugin, PostactionMixin, RegionMixin
+from pyfr.plugins.base import BasePlugin, CLIMixin, PostactionMixin, RegionMixin
+from pyfr.writers import BaseWriter, get_writer_by_name, get_writer_by_extn
 from pyfr.writers.native import NativeWriter
+from pyfr.util import subclasses
 
-
-class WriterPlugin(PostactionMixin, RegionMixin, BasePlugin):
+class WriterPlugin(CLIMixin, PostactionMixin, RegionMixin, BasePlugin):
     name = 'writer'
     systems = ['*']
     formulations = ['dual', 'std']
 
+    cli = [
```

Comment:
I'd leave the writer as-is for the moment.  As the new mechanism is a little bit confusing as the writer plugin is being used to export tavg files (which are produced by the tavg plugin).